plugins/chat/assets/javascripts/discourse/components/channels-list-direct.gjs

```diff
@@ -0,0 +1,215 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn, hash } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { schedule } from "@ember/runloop";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import concatClass from "discourse/helpers/concat-class";
+import noop from "discourse/helpers/noop";
+import dIcon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+import { bind } from "discourse-common/utils/decorators";
+import and from "truth-helpers/helpers/and";
+import not from "truth-helpers/helpers/not";
+import ChatModalNewMessage from "discourse/plugins/chat/discourse/components/chat/modal/new-message";
+import onResize from "../modifiers/chat/on-resize";
+import ChatChannelRow from "./chat-channel-row";
+
+export default class ChannelsListDirect extends Component {
+  @service chat;
+  @service chatStateManager;
+  @service chatChannelsManager;
+  @service site;
+  @service session;
+  @service modal;
+
+  @tracked hasScrollbar = false;
```

Comment:
I don't understand all the scrollbar logic, on my tests it's scrolling on the body so it doesn’t need to deal with all of this and this is what we should aim for:

![Screenshot 2024-01-11 at 19 28 34](https://github.com/discourse/discourse/assets/339945/231f184e-4f7a-4c75-b4a5-00affae7ac21)

Reply:
Yes let's remove all the useless stuff